build/elasticsearch/bin/docker-entrypoint.sh

```diff
@@ -1,6 +1,40 @@
 #!/bin/bash
+set -e
 
-# Run Elasticsearch and allow setting default settings via env vars
+# Files created by Elasticsearch should always be group writable too
+umask 0002
+
+# Create random user entry, if different from uid 1000, to support Openshift
+# https://docs.openshift.org/latest/creating_images/guidelines.html
+if ! whoami &> /dev/null; then
+    if [ -w /etc/passwd ]; then
+        echo "${USER_NAME:-default}:x:$(id -u):0:${USER_NAME:-default} user:/usr/share/elasticsearch:/sbin/nologin" >> /etc/passwd
+    fi
+fi
+
+run_as_other_user_if_needed() {
+    if [[ "$(id -u)" == "0" ]]; then
+        # If running as root, drop to specified UID and run command
+        exec chroot --userspec=1000 / "${@}"
+    else
+        # Either we are running in Openshift with random uid and are a member of the root group
+        # or with a custom --user
+        exec "${@}"
+    fi
+}
+
+# Allow user specify custom CMD, maybe bin/elasticsearch itself
+# for example to directly specify `-E` style parameters for elasticsearch on k8s
+# or simply to run /bin/bash to check the image
+if [[ "$1" != "eswrapper" ]]; then
+    if [[ "$(id -u)" == "0" ]] && [[ "$1" == *elasticsearch* ]]; then
+        exec chroot --userspec=1000 / "$@"
```

Comment:
Can we get a comment about this very unusual use of chroot? (It's very clever but very weird).

Reply:
Addressed in 9e3d43d20cbbeafea11a9fb4444cf27014255b90